packages/api/src/bsky-agent.ts

```diff
@@ -609,3 +645,103 @@ async function updateFeedPreferences(
   })
   return res
 }
+
+/**
+ * A helper specifically for updating muted words preferences
+ */
+async function updateMutedWords(
+  agent: BskyAgent,
+  mutedWords: AppBskyActorDefs.MutedWord[],
+  action: 'upsert' | 'update' | 'remove',
+) {
+  await updatePreferences(agent, (prefs: AppBskyActorDefs.Preferences) => {
+    let mutedWordsPref = prefs.findLast(
+      (pref) =>
+        AppBskyActorDefs.isMutedWordsPref(pref) &&
+        AppBskyActorDefs.validateMutedWordsPref(pref).success,
+    )
+
+    if (mutedWordsPref && AppBskyActorDefs.isMutedWordsPref(mutedWordsPref)) {
+      if (action === 'upsert' || action === 'update') {
+        outer: for (const newItem of mutedWords) {
+          for (const existingItem of mutedWordsPref.items) {
+            if (existingItem.value === newItem.value) {
+              existingItem.targets =
+                action === 'upsert'
+                  ? Array.from(
+                      new Set([...existingItem.targets, ...newItem.targets]),
+                    )
+                  : newItem.targets
+              continue outer
+            }
+          }
+
+          // never hit `continue`, add new item
+          if (action === 'upsert') {
+            mutedWordsPref.items.push(newItem)
+          }
+        }
+      } else if (action === 'remove') {
+        let i = mutedWordsPref.items.length
+
+        while (i--) {
+          const item = mutedWordsPref.items[i]
+
+          for (const word of mutedWords) {
+            if (item.value === word.value) {
+              if (word.targets.length) {
+                item.targets = word.targets
```

Comment:
this seems a little bit strange 🤔 

so if you're current prefs are is something like `{ word: 'blah', targets: ['tag', 'content'] }`, then if you do a `remove {word: 'blah', targets: ['tag']}`, it'll update the value to `{word: 'blah', targets: ['tag']}`?

feels like a strange syntax to me 🤔

Reply:
Actually just removed this, good catch, was a remnant of removing words with no targets. I decided to allow this bc it's simpler and allows to temp disable a word, basically.